flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/classification/SVM.scala

```diff
@@ -242,8 +275,21 @@ object SVM{
         }
       }
 
-      override def predict(value: T, model: DenseVector): Double = {
-        value.asBreeze dot model.asBreeze
+      override def predict(value: T, model: DenseVector, predictParameters: ParameterMap):
+        Double = {
+        val thresholdOption = predictParameters.get(Threshold)
+
+        val rawValue = value.asBreeze dot model.asBreeze
+        // If the Threshold option has been reset, we will get back a Some(None) thresholdOption
+        // causing the exception when we try to get the value. In that case we just return the
+        // raw value
+        try {
```

Comment:
This part is a bit hacky and will have a performance impact. However it allows to do what we want without needing to change the way ParameterMap works, and with the default behaviour being that the thresholdOption is set, the exception should be triggered most of the time.